Issues/91 vetting hcp users

cskeogh opened the pull request:
Implement future enhancements from PR #88. 

After registering, HCP accounts are locked awaiting vetting from the Slowvid administrators. To unlock account, use MongoDB Compass to set `locked` to `false`.

Lock accounts with too many incorrect password attempts. This is a security feature to mitigate against a brute force attack on the password.

Implement minimum password requirements.

Closes #91

cskeogh committed:
Lock accounts at register, require vetting #91

Lock accounts with incorrect password attempts

cskeogh committed:
Update README.md with unlock HCP account details #91

JoshuaBuditama reviewed (changes requested):
Code looks good and functional

Minor issues:
- Warning when compiling the front end of hcp (not a major issue since we don't seem to be building the hcp front end through GitHub Action)
![craig_pr#92](https://user-images.githubusercontent.com/26533936/97772599-909b2e00-1b98-11eb-8928-6cdabb711e1c.PNG)

- Can't register more than 1 account (not a major issue since we only need 1 account for the demo anyway). If there is an existing account the following error will be received.
![craig_pr#92part2](https://user-images.githubusercontent.com/26533936/97772597-8b3de380-1b98-11eb-96a6-bc865702757d.PNG)

- incorrectPasswordAttempt did not reset upon successful login

Major issue:
- a newly created hcp account had their "locked" attribute set to "true" as default.

Please prioritise addressing the major issue first.
Don't worry about the minor issues if there is not enough time to fix them.

Thanks Craig!

cskeogh committed:
Reset incorrect password attempt after successful login #91

cskeogh committed:
Remove unused variables #91

cskeogh commented:
> Warning when compiling the front end of hcp (not a major issue since we don't seem to be building the hcp front end through GitHub Action)

Thank you, now fixed.

> Can't register more than 1 account (not a major issue since we only need 1 account for the demo anyway). If there is an existing account the following error will be received.

This shouldn't happen. This PR has changed the database schema (HCPUser). Can you delete your `hcpusers` table in Mongo Compass and try again?

> incorrectPasswordAttempt did not reset upon successful login

Thank you, fixed.

> a newly created hcp account had their "locked" attribute set to "true" as default.

This is a security feature to prevent an attacker stealing a HCP's computer and immediately registering an account. This pull request's README.md states:

![Untitled](https://user-images.githubusercontent.com/5193990/97778568-f94dcf00-1bc7-11eb-825a-ab12b95782f1.png)
and I wrote in the pull request:
> After registering, HCP accounts are locked awaiting vetting from the Slowvid administrators.

JoshuaBuditama reviewed (approved):
Everything is fine now.
Ready to be merged.
Thanks @cskeogh !
